### ISSUE TITLE
IP Routing Card Not Loading Interfaces?

### ISSUE THREAD

**Issue by bt** [seud0nym/tch-gui-unhide#92]
I'm having an issue with adding static routes on the IP Routing card. When I add a route, I cannot select the interface because all the selections are blank.

<img width="1248" alt="2022-05-18_23-28-16" src="https://user-images.githubusercontent.com/1100843/169050525-f23c0184-b3ed-43e6-813e-98dfbf703668.png">

The selections can be selected, but they are just empty. I'm not exactly sure if they're associated to anything.

Looking at the HTML source, I can see:

```
<select id="Iface" class=" span2 centered" name="Iface"><option value="" selected="selected"></option><option value="" selected="selected"></option><option value="" selected="selected"></option><option value="" selected="selected"></option><option value="" selected="selected"></option></select>
```

**Comment by seud0nym** [comment]
Hmm. Not sure what caused it, but it was failing to extract the interface name for the transformer path.

If you are game, you can download the fixed version from https://raw.githubusercontent.com/seud0nym/tch-gui-unhide/master/build/lua/ansuel/iproutes/www/docroot/modals/iproutes-modal.lp and replace `/www/docroot/modals/iproutes-modal.lp`.

**Comment by bt** [comment]
Thanks for looking into this. I just tried the file that you provided and when I try to save, I get `An error occured while saving your changes.`.

**Comment by seud0nym** [comment]
What firmware version? 

Does `logread -e nginx` show any errors?

**Comment by bt** [comment]
Firmware version 17.2

Nothing in `logread`, nor in `/var/log/nginx/error.log`

**Comment by seud0nym** [comment]
Strange. I just tried it on both 17.2.20288 and 17.2.0468, and I was able to create a static route on both with no error.

That `An error occured while saving your changes` message is _really_ annoying - it give no indication of what or where the error is.

Maybe try restarting nginx `/etc/init.d/nginx restart` or even a reboot?

**Comment by seud0nym** [comment]
Did you have any luck saving your static route? I still can't reproduce an error...

**Comment by ilium007** [comment]
Happening here as well:

<img width="1179" alt="image" src="https://user-images.githubusercontent.com/732766/174428559-84bef279-7821-4f7b-b3ca-82b5a4f2395c.png">

**Comment by seud0nym** [comment]
> Happening here as well

Try downloading the fix as per comment https://github.com/seud0nym/tch-gui-unhide/issues/92#issuecomment-1135254504 above, and let me know how you go.

**Comment by ilium007** [comment]
That worked

**Comment by seud0nym** [comment]
This was fixed in Release 2022.06.01. Smart Modem Gen 3 users did not get an Update Available notification for that release because of a bug. If affected, please update manually.
